lib/src/rules/unrelated_type_equality_checks.dart

```diff
@@ -126,11 +129,19 @@ class DerivedClass2 extends ClassBase with Mixin {}
 
 ''';
 
+const String _dartCoreLibraryName = 'dart.core';
+
 bool _hasNonComparableOperands(BinaryExpression node) =>
     !DartTypeUtilities.isNullLiteral(node.leftOperand) &&
     !DartTypeUtilities.isNullLiteral(node.rightOperand) &&
     DartTypeUtilities.unrelatedTypes(
-        node.leftOperand.bestType, node.rightOperand.bestType);
+        node.leftOperand.bestType, node.rightOperand.bestType) &&
```

Comment:
Please extract `left` and `right` expressions; as well as `leftType` and `rightType`.
We reference them multiple times, and it hurts readability.

Reply:
Done.